bigquery_etl/cli/query.py

```diff
@@ -1282,36 +1337,64 @@ def initialize(name, sql_dir, project_id, dry_run):
     for query_file in query_files:
         sql_content = query_file.read_text()
 
-        # Enable init from query.sql files
-        # First deploys the schema, then runs the init
-        # This does not currently verify the accuracy of the schema
+        # Enable initialization from query.sql files
+        # Create the table by deploying the schema and metadata, then run the init.
+        # This does not currently verify the accuracy of the schema or that it
+        # matches the query.
         if "is_init()" in sql_content:
             project = query_file.parent.parent.parent.name
             dataset = query_file.parent.parent.name
             destination_table = query_file.parent.name
-            Schema.from_schema_file(query_file.parent / SCHEMA_FILE).deploy(
-                f"{project}.{dataset}.{destination_table}"
-            )
+            full_table_id = f"{project}.{dataset}.{destination_table}"
+
+            try:
+                client.get_table(full_table_id)
+                raise PreconditionFailed(
+                    f"Table {full_table_id} already exists. The initialization process is terminated."
+                )
+            except NotFound:
+                ctx.invoke(deploy, name=full_table_id, force=True)
+
+            query_file_path = Path(query_file)
+            metadata = Metadata.of_query_file(query_file_path)
+            if metadata and metadata.initialization:
```

Comment:
Retrieve initialization parameters from the table metadata. Currently supported for parallel initializaiton by sample_id.

Reply:
Updated.